### DISCUSSION TITLE
Conflicting watchOS library definitions

### DISCUSSION THREAD

**Discussion by mteahan85** [ge-org/multiplatform-swiftpackage#21] [Q&A]
Hello! I had a question about configurations for watchOS. I'm trying to use Kotlin multiplatform to build a framework. I'm trying to support all possible targets (iOS, tvOS, watchOS, macOS). I'm familiar with the Android, but new to the iOS side and trying to learn about Swift package manager. I'm having an issue building the Swift Package. 

I've set up my module's `build.gradle` file with all these targets and I've set up the plugin with versions for these targets:
```
multiplatformSwiftPackage {
    swiftToolsVersion("5.3")
    targetPlatforms {
        iOS { v("13") }
        tvOS { v("13")}
        watchOS { v("5")}
        macOS { v("10_15")}
    }
    outputDirectory(File(rootDir, "swiftpackage"))
    buildConfiguration { debug() }
    distributionMode { local() }
}
```

but I keep getting this error when running `./gradlew createSwiftPackage`:

```
> Task createXCFramework FAILED
Both watchos-armv7k and watchos-arm64_32 represent two equivalent library definitions.
```

 Has anyone run into this issue before or have thoughts on what I could be potentially be doing wrong?

**Comment by ge-org** [comment]
Hi @mteahan85,

unfortunately I'm not familiar with Apple watch development.
I did a bit of searching around but couldn't find a solution for this issue. There's a thread in the Apple dev forum but unfortunately without any answers: https://developer.apple.com/forums/thread/126219

You could try to build the framework with different settings to narrow down the problem or perhaps get a better error message.
- Only target watchos_arm64
    - replace `watchOS { v("5") }` with `targets("watchos_arm64", "watchos_x86", "watchos_x64") { v("5") }`
    - this removes support for watchos_arm32 aka watchos-armv7k
- Build the app in release mode
    - replace `buildConfiguration { debug() }` with `buildConfiguration { release() }`

You might also consider asking in the #multiplatform channel in the Kotlin Slack: https://surveys.jetbrains.com/s3/kotlin-slack-sign-up

**Reply by mteahan85** [reply]
Yeah, I saw that thread too and was disappointed it was never answered. Thanks for the tips, I'll see what happens when I try that out! 😄
